dartagnan/src/main/java/com/dat3m/dartagnan/program/event/Tag.java

```diff
@@ -327,4 +324,74 @@ public static String storeMO(String mo) {
             }
         }
     }
+
+    // =============================================================================================
+    // ========================================= Vulkan ============================================
+    // =============================================================================================
+    public static final class Vulkan {
+        // Scopes
+        public static final String SUB_GROUP = "SG";
+        public static final String WORK_GROUP = "WG";
+        public static final String QUEUE_FAMILY = "QF";
+        public static final String DEVICE = "DV";
+        public static final String PRIVATE = "PRIV";
```

Comment:
While you are defining the tag here, you are not assigning it to any event. I think you should remove it from here and define it in the cat file as `let PRIV = (R | W) \ NONPRIV` [following the alloy model.](https://github.com/KhronosGroup/Vulkan-MemoryModel/blob/master/alloy/spirv.als#L215)

Reply:
`PRIV` and `NONPRIV` are both used as explicit scopes in current grammar (and tests). To be consistent with the alloy model and their tests, only `NONPRIV `will be kept then.